content/tutorials/android-push-server-registration.textile

```diff
@@ -0,0 +1,239 @@
+ * In case your app requires access to FCM token as well, then override io.ably.lib.push.AblyFirebaseInstanceIdService#onTokenRefresh()
+ */
+public class AblyPushRegistrationService extends AblyFirebaseInstanceIdService {
+    @Override
+    public void onTokenRefresh() {
+        //Make sure to call super.onTokenRefresh to initialize Ably push environment.
+        super.onTokenRefresh();
+    }
+}
+```
+
+Next, in the same folder, that is 'server', add a new file and name it @AblyPushMessagingService.java@ and paste the following into it.
+
+```[java]
+package YOUR-PACKAGE-NAME;
+
+import io.ably.tutorial.push_tutorial_two.BuildConfig;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+import retrofit2.http.GET;
+import retrofit2.http.Query;
+
+public class ServerAPI {
+    public interface API {
+        @GET("/register")
+        Call<NetResponse> register(@Query("deviceId") String deviceId, @Query("registrationToken") String registrationToken, @Query("clientId") String clientId);
+    }
+
+
+    private static ServerAPI instance;
+
+    public static ServerAPI getInstance() {
+        if (instance == null) {
+            instance = new ServerAPI();
+        }
+        return instance;
+    }
+
+    private final API api;
+
+    private ServerAPI() {
+        Retrofit.Builder builder = new Retrofit.Builder();
+        builder.baseUrl(BuildConfig.ABLY_URL);
+        builder.addConverterFactory(GsonConverterFactory.create());
+
+        Retrofit retrofit = builder.build();
+        api = retrofit.create(API.class);
+    }
+
+    public API api() {
+        return api;
+    }
+}
+```
+
+If you observe, we have made use of the @NetResponse@ class. Let's go ahead and define this next, in the same file, i.e, 'server'. Create a new file named @NetResponse.java@ and paste the following into it:
+
+
+```[java]
+package YOUR-PACKAGE-NAME;
+
+import io.ably.lib.rest.DeviceDetails;
+
+public class NetResponse {
+    public String id;
+    public String platform;
+    public String formFactor;
+    public String clientId;
+    DeviceDetails.Push push;
+}
+
+```
+
+This class defines the data objects required to be sent to the server.
+
+h2(#step10). Step 10 - Testing the app
+
+<%= partial 'shared/tutorials/_testing_push_android' %>
+
+h2(#further-reading). Further Reading
+
+* Ably's "Push Notifications documentation":/general/push and info on "channels":/core-features/channels may be useful.
+
+* "FCM documentation":https://firebase.google.com/docs/cloud-messaging/ is a good place to find more information on the use of Push Notifications in Android.
+
+* You may also wish to deregister devices or get all of the stats related to your application implementing Push Notifications, such as the number and type of device registrations, etc. More information on how to do this can be found in the <a href="/general/push/admin">Push Admin API docs.</a>
```

Comment:
`<a href="/general/push/admin">Push Admin API docs.</a>` to `"Push Admin API docs":/general/push/admin`